templates/common/baremetal/files/baremetal-mdns-publisher.yaml

```diff
@@ -34,11 +34,23 @@ contents:
         - "-c"
         - |
           #/bin/bash
-          while [ "$(hostname)" == "$DEFAULT_LOCAL_HOSTNAME" ]
+          function get_hostname()
+          {
+            if [[ -s /etc/mdns/hostname.txt ]]; then
+              echo $(</etc/mdns/hostname.txt)
+            else
+              # if hostname wasn't updated by NM script, read hostname
+              hostname
+            fi
+          }
+          while [ "$(get_hostname)" == "$DEFAULT_LOCAL_HOSTNAME" ]
           do
             echo "hostname is still ${DEFAULT_LOCAL_HOSTNAME}"
             sleep 10
```

Comment:
I don't think it will kill us to try every second. It is literally just `cat`ting a file.